### PR TITLE
Fix: Button border radius 0 does not appear, and undefined border appears as 0

### LINKED ISSUE #22367: Border radius 0 in button block not working
**Describe the bug**
In the latest version, the button block is not able to set a square/rectanlge button, when border radius is set as 0, it results in a round corner instead of square/rectangle.

Previously, it will apply .no-border-radius class to button where it applies a border-radius: 0 css style, but the current version does not apply such CSS.

**To reproduce**
1. Add new button block
2. Change border settings, border radius to 0
3. Expect to see square/rectangle button

**Expected behavior**
When border-radius set as 0, expect to see a square/rectangle button instead of round corners.

**Screenshots**
Screen Shot 2020-05-14 at 1 46 53 PM
Captured from https://wordpress.org/gutenberg/

**Editor version (please complete the following information):**
- WordPress version: 5.3.2
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Default wordpress, no gutenberg plugin
- If the Gutenberg plugin is installed, which version is it? [e.g., 7.6]

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] All browsers

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Fixes: https://github.com/WordPress/gutenberg/issues/22367
This PR fixes a wrong CSS rule that made the border 0 radius in button not appear as expected.

It also fixes a bug, when the border-radius is not set (it is undefined and the theme value set with CSS is being used) we were displaying 0 as the border-radius by default, in that case, we should show the input empty.

## How has this been tested?
I enabled 2019 theme.
I added a button I set the border-radius to 0 and verified it appeared as expected on the editor and on the front end.
I added a button in 2019 (by default the button has some border) I verified the border did not appear as 0.

### CODE REVIEW

File: packages/block-library/src/button/style.scss [hunk 1]
```diff
@@ -43,7 +43,7 @@ $blocks-button__height: 56px;
 // the first selector is required for old buttons markup
 
 .wp-block-button.no-border-radius,
-.wp-block-button__link.wp-block-button.no-border-radius {
+.wp-block-button .wp-block-button__link.no-border-radius {
```
Comment (youknowriad) [inline packages/block-library/src/button/style.scss:46]: Can we remove `.wp-block-button`  here?

**Comment by youknowriad** [comment]
@ItsJonQ this seems a bit related to the "reset" behaviors we were discussing?

File: packages/block-library/src/button/style.scss (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/block-library/src/button/style.scss]: Suggestion applied :)

Comment (jorgefilipecosta) [reply packages/block-library/src/button/style.scss]: Suggestion applied :)

**Review by youknowriad — APPROVED** [review]

**Comment by ItsJonQ** [comment]
@jorgefilipecosta Ah! Thanks for catching this! I spotted a bug in the guard for the `floatClamp` util, which should tidy it up a bit.

Before:

```js
	if ( ! isFinite( value ) ) {
		return null;
	}
```

After:
```js
	if ( typeof value !== 'number' ) {
		return null;
	}
```

This should handle `""` values. I made that update on my end, and it looks like it's working.
I'll push it up. Would you be able to confirm?

Thank you! 🙏

**Comment by jorgefilipecosta** [comment]
Thank you for the enhancement @ItsJonQ. It worked great on my tests 👍

**Pull request merged**